packages/block-library/src/post-author/edit.js

```diff
@@ -1,25 +1,263 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
-import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import apiFetch from '@wordpress/api-fetch';
+import { useRef } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockColorsStyleSelector from './block-colors-selector';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+function PostAuthorDisplay( { props, author, authors } ) {
+	const ref = useRef();
+
+	const { editPost } = useDispatch( 'core/editor' );
+
+	const { fontSize, setFontSize } = props;
+	const {
+		TextColor,
+		BackgroundColor,
+		InspectorControlsColorPanel,
+		ColorPanel,
+	} = __experimentalUseColors(
+		[
+			{ name: 'textColor', property: 'color' },
+			{ name: 'backgroundColor', className: 'background-color' },
+		],
+		{
+			contrastCheckers: [
+				{
+					backgroundColor: true,
+					textColor: true,
+					fontSize: fontSize.size,
+				},
+			],
+			colorDetector: { targetRef: ref },
+			colorPanelProps: {
+				initialOpen: true,
+			},
+		},
+		[ fontSize.size ]
+	);
+
+	const { align, id, showAvatar, showBio, byline } = props.attributes;
+
+	const avatarSizes = [
+		{ value: 24, label: __( 'Small' ) },
+		{ value: 48, label: __( 'Medium' ) },
+		{ value: 96, label: __( 'Large' ) },
+	];
+
+	const changeAuthor = ( authorId ) => {
+		apiFetch( { path: '/wp/v2/users/' + authorId + '?context=edit' } ).then(
+			( newAuthor ) => {
+				editPost( { author: Number( authorId ) } );
+				props.setAttributes( {
+					id: newAuthor.id,
+					name: newAuthor.name,
+					avatarSize: props.attributes.avatarSize,
```

Comment:
Can this line be removed if it's not really changing avatarSize?

Reply:
I'll look into it.

Reply:
it's out!